### PR TITLE
Drop support for Django versions that have reached end of life

### PR DESCRIPTION
**Pull request by pamella**
**Description:** Describe in a couple of sentence what this PR adds

- Drop support for Django versions that have reached end of life.
  - https://www.djangoproject.com/download/
image
  - https://docs.djangoproject.com/en/5.0/faq/install/#what-python-version-can-i-use-with-django
image
- Upgrade dependencies.
- Fix tests running with tox in the GitHub action (and also rename its file to tests.yml)

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

N/A

**Merge deadline:** List merge deadline (if any)

N/A

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

- Tests should be running properly on GH actions.

[Screencast from 22-05-2024 16:14:07.webm](https://github.com/vintasoftware/drf-rw-serializers/assets/22326737/430bc2b4-e1b7-4e69-9fea-e6fb682f22bb)

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [X] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: .github/workflows/tests.yml
Comment (pamella) [inline .github/workflows/tests.yml:1]: - Rename to `tests`.
- Now runs on every push and PR.

File: Makefile [hunk 1]
```diff
@@ -49,7 +49,6 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
 	pip-compile --upgrade -o requirements/doc.txt requirements/base.in requirements/doc.in
 	pip-compile --upgrade -o requirements/quality.txt requirements/quality.in
 	pip-compile --upgrade -o requirements/test.txt requirements/base.in requirements/test.in
-	pip-compile --upgrade -o requirements/travis.txt requirements/travis.in
```
Comment (pamella) [inline Makefile:52]: Drive-by fix.

**Review by fjsj — APPROVED** [review]
LGTM, thanks!
